listenbrainz/webserver/static/js/jsx/follow-users.jsx

```diff
@@ -42,6 +46,43 @@ export class FollowUsers extends React.Component {
     });
   }
 
+  saveFollowList(event) {
+    var listName = this.state.listName;
+    if (this.nameInput.value != undefined && this.nameInput.value != "") {
+      listName = this.nameInput.value;
+    }
+    fetch(this.state.saveUrl, {
+      method: "POST",
+      body: JSON.stringify({
+        "users": this.state.users,
+        "name": listName,
+        "id": this.state.listId,
+      }),
+    })
+    .then(response => response.json())
+    .then(data => {
+      console.debug(data);
+      console.debug("old List ID: " + this.state.listId);
+      this.setState(prevState => {
+        return {listId: data.list_id};
+      });
+      console.debug("new List ID: " + this.state.listId);
```

Comment:
This console log won't necessarily work because setState is asynchronous.

You can pass an optional callback to setState as a second argument, and in that callback you can be sure the value has been updated. That's where you'll want that new List ID log.

Reply:
whoops.